### PR TITLE
Implemented typedJs, refactored code into modules, created db connect…

### PR DESCRIPTION
**Pull request by IsraelGboluwaga**
…ion error check, started auth page

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -5,19 +5,27 @@ const logger = require('morgan');
...
 
+const appRoutes = require('./routes/index');
+const activity = require('./routes/activity');
+const login = require('./routes/login');
+const logout = require('./routes/logout');
 
-const appRoutes = require('./routes/app');
-
+require('dotenv').config();
 let app = express();
```
Comment (mayomi1) [inline app.js]: const

File: app.js [hunk 2]
```diff
@@ -5,19 +5,27 @@ const logger = require('morgan');
...
 
 //Set db connection
-mongoose.connect(db.url);
+mongoose.connect(process.env.DB_URL);
+mongoose.Promise = global.Promise;
+let db_connection = mongoose.connection;
```
Comment (mayomi1) [inline app.js]: const

File: config/utils.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const constants = require('./constants');
+let regex, regObj;
+
+let check = {
```
Comment (mayomi1) [inline config/utils.js]: const

File: models/user.js [hunk 4]
```diff
@@ -1,16 +1,53 @@
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 const uniqueValidator = require('mongoose-unique-validator');
+const bcrypt = require('bcrypt');
+const constants = require('../config/constants');
 
 
 let userModel = new Schema({
```
Comment (mayomi1) [inline models/user.js]: const

File: public/javascripts/scripts.js [hunk 5]
```diff
@@ -1,17 +1,20 @@
+setTimeout(function () {
+    $("#error-div").hide(500).empty();
+}, 2500);
 
 let options = {
```
Comment (mayomi1) [inline public/javascripts/scripts.js]: const

File: routes/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+const express = require('express');
+const router = express.Router();
+
+const User = require('../models/user');
+const constants = require('../config/constants');
+const check = require('../config/utils');
+const appName = constants.APP_NAME;
+
+let pass = {
```
Comment (mayomi1) [inline routes/index.js]: const

and it will be good if you can have a better name for this pass

File: routes/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+/* GET home page for unregistered users */
+router.get('/', function (req, res, next) {
+    res.render('index', pass);
+});
+
+//User's sign up
+router.post('/', (req, res, next) => {
+    let username = req.body.username;
```
Comment (mayomi1) [inline routes/index.js]: all to const

File: routes/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    let password = req.body.password;
+    let email = req.body.email;
+    let phone = req.body.phone;
+
+
+    if (!username && !password && !email && !phone) {
+        pass.error_message = constants.EMPTY_PARAMS;
+        res.redirect('/')
```
Comment (mayomi1) [inline routes/index.js]: missing return

File: routes/index.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    if (!check.email(email)) {
+        pass.error_message = constants.INVALID_EMAIL;
+        res.redirect('/')
```
Comment (mayomi1) [inline routes/index.js]: missing return

File: routes/index.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    if (!check.username(username)) {
+        pass.error_message = constants.INVALID_USERNAME;
+        res.redirect('/')
```
Comment (mayomi1) [inline routes/index.js]: missing return

File: routes/index.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    if (!check.phone(phone)) {
+        pass.error_message = constants.INVALID_PHONE;
+        res.redirect('/')
```
Comment (mayomi1) [inline routes/index.js]: missing return

File: routes/index.js [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    let params = {
```
Comment (mayomi1) [inline routes/index.js]: const

File: routes/index.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+        username: username,
+        password: password,
+        email: email,
+        phone: phone
+    };
+
+    let newUser = new User(params);
```
Comment (mayomi1) [inline routes/index.js]: const

File: routes/index.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+    newUser.save((err, user, next) => {
+        if (err) {
+            console.log(err);
```
Comment (mayomi1) [inline routes/index.js]: throw this error or do `res.json()`

File: routes/index.js [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+            return next(err);
+        }
+
+        if (user)
+            req.session = {
```
Comment (mayomi1) [inline routes/index.js:64]: why not , `req.session.user = user._id`

**Review by mayomi1 — COMMENTED** [review]
change all the let to const, and I cant find where you are rendering the view yet

File: routes/index.js (hunk 9 above)
Comment (IsraelGboluwaga) [reply routes/index.js]: Why should there be a return if it's redirecting? @mayomi1

File: routes/index.js (hunk 15 above)
Comment (IsraelGboluwaga) [reply routes/index.js:64]: An error was thrown for `session` not being defined. That's why I chose an object

**Pull request merged**